src/components/pagination/pager-pagination.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {Fragment, useEffect, useState} from "react";
+import {Pagination} from "react-bootstrap";
+import {PagerItem} from "./pager-item";
+
+export interface PaginationProps {
+    numberOfPageButtonsToShowAfterAndBeforeCurrent: number
+    onPageChange: (pageIndex: number) => void
+    lastPageIndex: number
+}
+
+export const PagerPagination: React.FC<PaginationProps> = ({numberOfPageButtonsToShowAfterAndBeforeCurrent, onPageChange, lastPageIndex}) => {
+    if (numberOfPageButtonsToShowAfterAndBeforeCurrent % 2 !== 0) {
+        throw new Error("number of pages to show must be even!")
+    }
+
+    const [pageIndex, setPageIndex] = useState(0);
+    const correctedPageIndex = Math.min(pageIndex, lastPageIndex);
+    const wantedUpperPageIndex = correctedPageIndex + numberOfPageButtonsToShowAfterAndBeforeCurrent;
+    const wantedLowerPageIndex = correctedPageIndex - numberOfPageButtonsToShowAfterAndBeforeCurrent;
+
+    useEffect(() => {
+        onPageChange(pageIndex)
+    }, [onPageChange, pageIndex])
```

Comment:
you don't need to put `onPageChange` in the `[]`
```suggestion
    useEffect(() => {
        onPageChange(pageIndex)
    }, [pageIndex])
```

Reply:
ESLint says: I do

Reply:
interesting, well then nevermind